### PR TITLE
adding in inventory to unified job query for analytics

### PR DESCRIPTION
**Pull request by rebeccahhh**
##### SUMMARY

adds inventory name and ID to collector

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 14.0.0
```

### CODE REVIEW

**Comment by wenottingham** [comment]
cc @Ladas - this will need processor support once it lands

**Comment by rooftopcellist** [comment]
@rebeccahhh it looks like this test might need to be updated based on the test failure - https://github.com/ansible/awx/blob/b69b53f52774784cf88816af507c65b96c79c6bd/awx/main/tests/functional/analytics/test_collectors.py#L70

**Comment by Ladas** [comment]
it will need to join to `main_inventory` table most likely

**Comment by rebeccahhh** [comment]
@rooftopcellist actually @Ladas is right, we're going to have to do a table update. I opened this PR quite prematurely :sweat_smile: 
Working on that now.

**Comment by rebeccahhh** [comment]
Alright updated, should be passing that test now @rooftopcellist

File: awx/main/analytics/collectors.py [hunk 1]
```diff
@@ -262,6 +264,7 @@ def _copy_table(table, query, path):
                               main_jobevent.event_data::json->'res'->'warnings' AS warnings,
                               main_jobevent.event_data::json->'res'->'deprecations' AS deprecations
                               FROM main_jobevent 
+                              LEFT JOIN main_inventory ON main_inventory.id = main_job.inventory_id
```
Comment (jbradberry) [inline awx/main/analytics/collectors.py]: This seems to be a typo, it looks like we're dealing with `main_jobevent` in this query

File: awx/main/analytics/collectors.py [hunk 2]
```diff
@@ -287,8 +292,9 @@ def _copy_table(table, query, path):
                                  main_unifiedjob.elapsed, 
                                  main_unifiedjob.job_explanation, 
                                  main_unifiedjob.instance_group_id
-                                 FROM main_unifiedjob
+                                 FROM main_unifiedjob, main_job
```
Comment (jbradberry) [inline awx/main/analytics/collectors.py]: I think this isn't the type of join we want, this is a type of implicit inner join called an equi join.

Comment (rebeccahhh) [reply awx/main/analytics/collectors.py]: wouldn't that just require changing the left join below, not this line?

File: awx/main/analytics/collectors.py (hunk 1 above)
Comment (rebeccahhh) [reply awx/main/analytics/collectors.py]: ope, that's what I get for typing too fast, :+1:

File: awx/main/analytics/collectors.py (hunk 2 above)
Comment (rebeccahhh) [reply awx/main/analytics/collectors.py]: :facepalm: nvm

**Review by jbradberry — APPROVED** [review]
This looks ok to me now

File: awx/main/analytics/collectors.py [hunk 3]
```diff
@@ -254,6 +254,8 @@ def _copy_table(table, query, path):
                               main_jobevent.task,
                               main_jobevent.role, 
                               main_jobevent.job_id, 
+                              main_jobevent.inventory_id,
```
Comment (rooftopcellist) [inline awx/main/analytics/collectors.py]: I think you need `main_inventory.id,` here instead.

Comment (rebeccahhh) [reply awx/main/analytics/collectors.py]: I actually need main_job.inventory_id so that I can match it to the inventory table, should be fixed

**Comment by rebeccahhh** [comment]
@Ladas @notting merging this today is likely

**Review by beeankha — APPROVED** [review]

**Review by wenottingham — APPROVED** [review]

**Comment by kdelee** [comment]
@rebeccahhh in the future you can run YOLO with TESTEXPR=analytics

need to update a test because of the version change for the unified jobs csv

**Comment by rebeccahhh** [comment]
@kdelee good to know, I was on PTO when this merged, I'll work with @rooftopcellist to fix this. :smile_cat: 
update--should be taken care of

**Pull request merged**